rotkehlchen/tests/db/test_history_events.py

```diff
@@ -311,3 +312,172 @@ def test_delete_last_event(database):
+        ), HistoryEvent(
+            event_identifier=event_identifiers[3],
+            sequence_index=2,
+            timestamp=TimestampMS(4000),
+            location=Location.COINBASE,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(1000)),
+        ),
+    )
+    with database.conn.write_ctx() as write_cursor:
+        history_events.add_history_events(  # add dummy history events
+            write_cursor=write_cursor,
+            history=dummy_events,
+        )
+
+    with database.conn.read_ctx() as cursor:
+        # free and premium results should be same with len(events) < FREE_LIMIT
+        for filters in (  # trying different filters with some combinations
+            HistoryEventFilterQuery.make(location=Location.OPTIMISM),
+            HistoryEventFilterQuery.make(assets=(A_ETH,)),
+            HistoryEventFilterQuery.make(assets=(A_ETH,), location=Location.COINBASE),
+            HistoryEventFilterQuery.make(from_ts=Timestamp(2)),
+            HistoryEventFilterQuery.make(assets=(A_USDT,), to_ts=Timestamp(3)),
+            HistoryEventFilterQuery.make(exclude_ignored_assets=True),
+        ):
+            assert history_events.get_history_events(  # when grouping
+                cursor=cursor,
+                filter_query=filters,
+                has_premium=True,
+                group_by_event_ids=True,
+            ) == history_events.get_history_events(
+                cursor=cursor,
+                filter_query=filters,
+                has_premium=False,
+                group_by_event_ids=True,
+            ) != []
+            assert history_events.get_history_events(  # when not grouping
+                cursor=cursor,
+                filter_query=filters,
+                has_premium=True,
+                group_by_event_ids=False,
+            ) == history_events.get_history_events(
+                cursor=cursor,
+                filter_query=filters,
+                has_premium=False,
+                group_by_event_ids=False,
+            ) != []
+
+        with patch.object(target=history_events, attribute='free_history_events_limit', new=2):
+            assert history_events.free_history_events_limit == 2
+            for filters in (  # trying different filters with some combinations
+                HistoryEventFilterQuery.make(event_types=[HistoryEventType.RECEIVE]),
+                HistoryEventFilterQuery.make(assets=(A_ETH, A_USDT)),
+                HistoryEventFilterQuery.make(location=Location.COINBASE),
+                HistoryEventFilterQuery.make(from_ts=Timestamp(2)),
+                HistoryEventFilterQuery.make(assets=(A_ETH, A_USDC), from_ts=Timestamp(2)),
+                HistoryEventFilterQuery.make(exclude_ignored_assets=True),
+            ):
+                premium_grouped_result = history_events.get_history_events(  # when grouping
```

Comment:
what are you checking here? changing the free history event limit?

Reply:
This checks that when results of a query hits the free limit, then we should not return more the FREE_LIMIT results. `len(free_grouped_result) == min(2, len(premium_grouped_result))`
And the N limited results that are returned, are the first N results returned in premium query with the same filters. `free_grouped_result == premium_grouped_result[:2]`